src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -194,6 +190,31 @@ def _validate_target_field(
         default=None,
     )
 
+    mixins: Optional[Dict[str, Dict]] = Field(
```

Comment:
Currently i use plain dict, as mixins are used before validation, when we manage the raw dict data. 
It's simple, but any misspellings will pas unnoticed.
Another option is, as we have to include any posible field in every entity, to create a class in runtime.  We can get all entites,  extract their fields and create a pydantic class with all those fields set to optional. 
It would allow us to spot any mistaken key in mixins.